static/js/search_suggestion.js

```diff
@@ -365,7 +365,8 @@ function get_special_filter_suggestions(last, operators, suggestions) {
         suggestions = _.map(suggestions, function (suggestion) {
             return {
                 search_string: '-' + suggestion.search_string,
-                description: 'exclude ' + suggestion.description,
+                description: i18n.t('exclude __description__',
+                                    {description: suggestion.description}),
```

Comment:
So, the issue of negating descriptions is this: Is this the right way to internationalize excluding a suggestion, or do we want to just have an `excluded_description` field to handle cases of that so that translators can write somewhat differently?

What's challenging is this -- you don't want translators taking `All messages`, translating to their language's version of `All messages` with that capitalization, and then having us render "Exclude All messages"; we want "Exclude all messages" because that's reasonable case.  

Ideally, our code would handle the capitalization-of-sentence piece in some i18n-correct way, and we'd just need to change various the strings below to e.g. "all messages" to make this work.  

If we can't make that happen, the alternative is to just add a `negated_description: i18n.t('Exclude all messages')`.  But that's clearly a worse experience for translators, so we should only do that if we can't make the above work.

https://www.i18next.com/ may have some info.

Reply:
i18next doesn’t have any case-switching functionality as far as I can tell. There are some [language issues to watch out for](https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform), and [users who complain when developers get it wrong](https://twitter.com/razmadzekoko/status/1060144235461201920); I haven’t found a JS library that tries to get it right.

Also, imagining a hypothetical language whose word order is the reverse of English, we wouldn’t want the description of `-stream:foo` capitalized from “foo stream exclude” to “Foo stream exclude”. I guess we could work around that problem by
1. substituting a *template* into `exclude __description__` (`__description__ exclude`), yielding `exclude stream __stream__` (`__stream__ stream exclude`),
2. capitalizing that to `Exclude stream __stream__` (`__stream__ stream exclude`),
3. doing another round of substitution.

But the `negated_description` plan is sounding much simpler.

Reply:
Yeah, the concern I had with `negated_description` is that I'm not sure there's a good way for that to internationalize capitalization for the overall setup, which I think currently makes a comma-separated list of these things, with only the first capitalized?

![image](https://user-images.githubusercontent.com/2746074/70565270-9504c580-1b46-11ea-9c50-856bfc6dad9a.png)

The right answer might be just to remove capitalization, and then we probably want `negated_description` as well, so we don't need to worry about word order?